pkg/server/plugin/datastore/sql/sql.go

```diff
@@ -1161,3 +1160,14 @@ func newPlugin() *sqlPlugin {
 func New() datastore.Plugin {
 	return newPlugin()
 }
+
+func stringsToColumn(ss []string) string {
+	return strings.Join(ss, ",")
+}
+
+func columnToStrings(s string) []string {
+	if s == "" {
+		return nil
+	}
+	return strings.Split(s, ",")
```

Comment:
`FederatesWith []string` is not a valid field type for use with GORM (you need a slice of models). We could model a one-to-many by creating a new struct for the federated id and doing a slice of those. Or we can do many-to-many with full back-reference support (and avoid AutoMigrate because it fails trying to create the join table twice). Or we can do many-to-many without back-reference support.  The "query by federated trust id" is an optimization could be useful. It would certainly be more performant with a many-to-many relationship set up, but nothing precludes doing a LIKE expression against the currently implemented federates_with column. We could also press for GORM to fix the auto migrate bug....

Reply:
Welp. Consider me embarrassed. The many2many auto migrate bug is fixed in the latest GORM. I'll go back to many2many and update the gorm package.